### PR TITLE
Implement missing 3DS2 features according to API reference

### PR DESCRIPTION
**Pull request by haydnchapman**
Implement 3D Secure 2.0 according to the updated API reference (https://docs.adyen.com/developers/risk-management/3d-secure-2-0/server-integration/api-reference-3d-secure-2-0)

Changes include:
- Add missing properties to `BrowserInfo` object
- Add Device Channel, Device Fingerprint complete Indicator and Transaction Result status enumerations
- Add Authorise3DS2 methods to Payment service 
- Add `IdentifyShopper` and `ChallengeShopper` constants to the authorisation ResultCode enum

### CODE REVIEW

**Comment by AlexandrosMor** [comment]
Hey @haydnchapman, 

Great job for adding 3ds to the library. I would like to ask you if it is possible to create some mocking tests similar what we have so far.

Thank you in advance, 
Alexandros
Adyen

**Comment by haydnchapman** [comment]
Hi @AlexandrosMor - no problem, I will implement them shortly.

Thanks,
Haydn

**Comment by haydnchapman** [comment]
@AlexandrosMor - tests following the existing approach have been added.

Cheers

**Comment by haydnchapman** [comment]
@AlexandrosMor - Hi Alexandros,

Are there any further changes that you require for this PR?

Thanks,
Haydn

File: Adyen.EcommLibrary/Model/Checkout/PaymentResultResponse.cs [hunk 1]
```diff
@@ -66,6 +66,7 @@ public enum ResultCodeEnum
             /// </summary>
             [EnumMember(Value = "RedirectShopper")]
             RedirectShopper = 7
+
```
Comment (AlexandrosMor) [inline Adyen.EcommLibrary/Model/Checkout/PaymentResultResponse.cs]: Please remove the empty line

File: Adyen.EcommLibrary/Service/Resource/Payment/Authorise.cs [hunk 2]
```diff
@@ -11,7 +11,9 @@ public Authorise(AbstractService abstractService)
                     "merchantAccount",
                     "amount.value",
                     "amount.currency",
-                    "reference"
+                    "reference",
+                    "threeDS2RequestData.deviceChannel",
```
Comment (AlexandrosMor) [inline Adyen.EcommLibrary/Service/Resource/Payment/Authorise.cs]: Those are not required fields. The list contains only required fields

Comment (haydnchapman) [reply Adyen.EcommLibrary/Service/Resource/Payment/Authorise.cs]: Removed on-required fields

File: Adyen.EcommLibrary/Model/Checkout/PaymentResultResponse.cs (hunk 1 above)
Comment (haydnchapman) [reply Adyen.EcommLibrary/Model/Checkout/PaymentResultResponse.cs]: empty line removed

**Pull request merged**